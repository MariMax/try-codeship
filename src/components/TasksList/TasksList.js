// TasksList
import React, { PropTypes } from 'react';
import TaskItem from './components/TaskItem/TaskItem';

function TasksList(props){
    return (
      <ul className="list-group pre-scrollable tasks-list">
        {props.list.map((task) => <TaskItem key={task.id} task={task} onDoneChange={props.onDoneChange}/>)}
      </ul>
    );
}

TasksList.propTypes = {
  list : PropTypes.array.isRequired,
  onDoneChange: PropTypes.func.isRequired
};

export default TasksList;
