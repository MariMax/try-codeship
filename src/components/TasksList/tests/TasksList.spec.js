import React from 'react';
import renderer from 'react-test-renderer'

import TasksList from '../TasksList'
var tasksListMock = [
    {
        id: 1,
        title: 'task 1',
        done: false
    },
    {
        id: 2,
        title: 'task 2',
        done: true
    },
    {
        id: 3,
        title: 'task 3',
        done: true
    }
];

describe('TasksList component', function(){

    it('should render tasks list', function(){
        const component = renderer.create(
            <TasksList list={tasksListMock} onDoneChange={function(){}}/>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});