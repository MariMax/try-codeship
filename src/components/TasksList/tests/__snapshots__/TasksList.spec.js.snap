exports[`TasksList component should render tasks list 1`] = `
<ul
  className="list-group pre-scrollable tasks-list">
  <li
    className="list-group-item d-flex flex-items-xs-middle flex-items-xs-between">
    <div>
      task 1
    </div>
    <div>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox" />
      <a
        className="btn btn-secondary m-l-1"
        onClick={[Function]}
        style={Object {}}>
        edit
      </a>
    </div>
  </li>
  <li
    className="list-group-item d-flex flex-items-xs-middle flex-items-xs-between">
    <div>
      task 2
    </div>
    <div>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox" />
      <a
        className="btn btn-secondary m-l-1"
        onClick={[Function]}
        style={Object {}}>
        edit
      </a>
    </div>
  </li>
  <li
    className="list-group-item d-flex flex-items-xs-middle flex-items-xs-between">
    <div>
      task 3
    </div>
    <div>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox" />
      <a
        className="btn btn-secondary m-l-1"
        onClick={[Function]}
        style={Object {}}>
        edit
      </a>
    </div>
  </li>
</ul>
`;
